@xen-orchestra/backups/writers/_MixinBackupWriter.js

```diff
@@ -52,12 +52,20 @@ exports.MixinBackupWriter = (BaseClass = Object) =>
     }
 
     async afterBackup() {
-      const { disableMergeWorker } = this._backup.config
+      const { disableMergeWorker, disableCachePregeneration } = this._backup.config
       // merge worker only compatible with local remotes
       const { handler } = this._adapter
       const willMergeInWorker = !disableMergeWorker && typeof handler._getRealPath === 'function'
 
       const { merge } = await this._cleanVm({ remove: true, merge: !willMergeInWorker })
+
+      // cache should be refreshed since we added on new backup
+      // and also cleanVm may have deleted brokenc json
+      await this.invalidateVmBackupListCache(this._backup.vm.uuid)
+
+      if (!disableCachePregeneration) {
```

Comment:
I think I prefer having it disabled by default and enabling it via the config.

Also, what do you think about the name `regenerateCacheAfterbackup`?

I'm still ill-at-ease with this approach and the a bit afraid of the perf impact to enabling it by default in production. I would prefer more targeted updates, like removing entries in `RemoteAdapter`'s deletion methods and adding the new entry after a backup. I feel like this would not be too difficult and it would bring a real performance improvements.

What do you think?